src/rich_click/rich_click.py

```diff
@@ -99,54 +106,52 @@
 # Define sorted groups of panels to display subcommands
 COMMAND_GROUPS: Dict[str, List[Dict[str, Union[str, List[str]]]]] = {}
 # Define sorted groups of panels to display options and arguments
-OPTION_GROUPS: Dict[str, List[Dict[str, Union[str, List[str]]]]] = {}
+OPTION_GROUPS: Dict[str, List[Dict[str, Union[str, List[str], Dict[str, List[str]]]]]] = {}
 USE_CLICK_SHORT_HELP = False  # Use click's default function to truncate help text
 
+highlighter = OptionHighlighter()
+_formatter = None
 
-# Rich regex highlighter
-class OptionHighlighter(RegexHighlighter):
-    """Highlights our special options."""
 
-    highlights = [
-        r"(^|\W)(?P<switch>\-\w+)(?![a-zA-Z0-9])",
-        r"(^|\W)(?P<option>\-\-[\w\-]+)(?![a-zA-Z0-9])",
-        r"(^|\W)(?P<argument>[A-Z0-9\_]+)(?![_a-zA-Z0-9])",
-        r"(?P<metavar>\<[^\>]+\>)",
-        r"(?P<usage>Usage: )",
-    ]
+def _get_rich_formatter(formatter: Optional[click.HelpFormatter] = None) -> RichHelpFormatter:
```

Comment:
This is possibly not needed since the formatter entry points are in the public functions. But, I was a bit unsure of other ways users may be using the "private-by-convention" functions in this module.

My thought was maybe someone is calling these internal functions to supplement behavior, and are calling them without the newly added `formatter` argument. In that case, this should fall back to the original behavior since it will then resolve the formatter from the active click context or the module-level configuration.